.github/workflows/check-replays.yml

```diff
@@ -0,0 +1,206 @@
+name: check-replays
+
+permissions:
+  contents: read
+  pull-requests: write
+
+on:
+  workflow_call:
+    inputs:
+      game:
+        required: true
+        type: string
+        description: "Game to check (only GeneralsMD for now)"
+      preset:
+        required: true
+        type: string
+        description: "CMake preset"
+
+jobs:
+  build:
+    name: ${{ inputs.preset }}
+    runs-on: windows-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Download Game Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.game }}-${{ inputs.preset }}
+          path: build
+
+      - name: Cache Game Data
+        id: cache-gamedata
+        uses: actions/cache@v4
+        with:
+          path: C:\GameData
+          key: gamedata-permanent-cache-v2
+
+      - name: Download Game Data from Cloudflare R2
+        if: ${{ steps.cache-gamedata.outputs.cache-hit != 'true' }}
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.R2_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.R2_SECRET_ACCESS_KEY }}
+          AWS_ENDPOINT_URL: ${{ secrets.R2_ENDPOINT_URL }}
+          EXPECTED_HASH: "3BE86F84B1E771BEA01BD2DBA39442683381AE4401716E9AE9D4AD95C8F45372"
+        shell: pwsh
+        run: |
+          # Download trimmed gamedata of both ZH and Generals. This data cannot be used for playing because it's
```

Comment:
"Generals 1.08 and Generals Zero Hour 1.04"

I suggest to always name Generals first for consistency.